test/upgrade_test.go

```diff
@@ -0,0 +1,246 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"github.com/tektoncd/pipeline/test/parse"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	knativetest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/helpers"
+)
+
+var (
+	filterOwnerReference              = cmpopts.IgnoreFields(metav1.OwnerReference{}, "UID", "Controller", "BlockOwnerDeletion")
+	filterV1SidecarState              = cmpopts.IgnoreFields(v1.SidecarState{}, "ImageID")
+	filterVolumeMountsName            = cmpopts.IgnoreFields(corev1.VolumeMount{}, "Name")
+	filterTaskRunStatusProvenance     = cmpopts.IgnoreFields(v1.TaskRunStatusFields{}, "Provenance")
+	filterPipelineRunStatusProvenance = cmpopts.IgnoreFields(v1.PipelineRunStatusFields{}, "Provenance")
+)
+
+// TestUpgradeAgainstPreviousVersion creates a pipelineRun with a basic
+// TaskRun and a basic Pipeline and examines the events and runs created
+// are successful.
+func TestUpgradeAgainstPreviousVersion(t *testing.T) {
```

Comment:
It might help to separate the pipelinerun and taskrun tests; it would be clearer what each test is intended to test.

Reply:
I break them into TestSimpleTaskRun and TestSimplePipelineRun